torch/_functorch/aot_autograd.py

```diff
@@ -1659,8 +1659,27 @@ def _collect_assigned_tensor_attributes(kp, v, _v):
                 # TODO(avik): Assigning all other types are allowed right now.
                 # Maybe in the future we want to limit this to primitive types?
 
+        new_attrs = _get_attributes(mod)
+        if len(new_attrs) != len(snapshot):
+            added_attrs = new_attrs.keys() - snapshot.keys()
+            deleted_attrs = snapshot.keys() - new_attrs.keys()
+
+            if len(added_attrs) > 0:
+                raise ValueError(
+                    f"During export following attrs were created: {added_attrs} "
```

Comment:
I think there are two things here that seem ambiguous:

(1) whether or not the user puts an attribute on their `nn.Module` at init time, or inside of their model forward

(2) whether or not the attribute that the user sets is properly a buffer

I think this PR is mainly addressing (1). But is the buffer part necessary? We should probably make both of these explicit in the message (they should set the attribute at model init time and not in the forward, **and** they should set it through the `register_buffer` API)

Reply:
Sounds good!